modules/library/main/src/main/java/org/geotools/filter/visitor/BindingFilterVisitor.java

```diff
@@ -0,0 +1,322 @@
+
+        return getFactory(extraData).before(expr1, expr2, before.getMatchAction());
+
+    }
+
+    public Object visit(Begins begins, Object extraData) {
+        Expression expr1 = optimizeTime(begins.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(begins.getExpression2(), extraData);
+
+        return getFactory(extraData).begins(expr1, expr2, begins.getMatchAction());
+    };
+
+    public Object visit(BegunBy begunBy, Object extraData) {
+        Expression expr1 = optimizeTime(begunBy.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(begunBy.getExpression2(), extraData);
+
+        return getFactory(extraData).begins(expr1, expr2, begunBy.getMatchAction());
+    }
+
+    public Object visit(During during, Object extraData) {
+        Expression expr1 = optimizeTime(during.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(during.getExpression2(), extraData);
+
+        return getFactory(extraData).during(expr1, expr2, during.getMatchAction());
+    }
+
+    public Object visit(EndedBy endedBy, Object extraData) {
+        Expression expr1 = optimizeTime(endedBy.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(endedBy.getExpression2(), extraData);
+
+        return getFactory(extraData).endedBy(expr1, expr2, endedBy.getMatchAction());
+    }
+
+    public Object visit(Ends ends, Object extraData) {
+        Expression expr1 = optimizeTime(ends.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(ends.getExpression2(), extraData);
+
+        return getFactory(extraData).ends(expr1, expr2, ends.getMatchAction());
+    }
+
+    public Object visit(Meets meets, Object extraData) {
+        Expression expr1 = optimizeTime(meets.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(meets.getExpression2(), extraData);
+
+        return getFactory(extraData).meets(expr1, expr2, meets.getMatchAction());
+    }
+
+    public Object visit(MetBy metBy, Object extraData) {
+        Expression expr1 = optimizeTime(metBy.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(metBy.getExpression2(), extraData);
+
+        return getFactory(extraData).metBy(expr1, expr2, metBy.getMatchAction());
+    }
+
+    public Object visit(OverlappedBy overlappedBy, Object extraData) {
+        Expression expr1 = optimizeTime(overlappedBy.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(overlappedBy.getExpression2(), extraData);
+
+        return getFactory(extraData).overlappedBy(expr1, expr2, overlappedBy.getMatchAction());
+    }
+
+    public Object visit(TContains contains, Object extraData) {
+        Expression expr1 = optimizeTime(contains.getExpression1(), extraData);
+        Expression expr2 = optimizeTime(contains.getExpression2(), extraData);
+
+        return getFactory(extraData).contains(expr1, expr2, contains.getMatchAction());
```

Comment:
Aha! Found it. This statement should use the method "tcontains" not "contains". Changing this method to "tcontains" fixes the build failure in gt-jdbc-postgis online tests.